### PR TITLE
fix(ui): prevent recursive flow calls

### LINKED ISSUE #1474: Infinite loop with subflow
### Expected Behavior

Created a recursive flow create an infinite loop :

```
id: "flow-a"
namespace: "dev"
tasks:
  - id: log
    type: io.kestra.core.tasks.log.Log
    message: hello
  - id: subflow
    type: io.kestra.core.tasks.flows.Flow
    flowId: flow-a
    namespace: dev
```

### Actual Behaviour

_No response_

### Steps To Reproduce

_No response_

### Environment Information

- Kestra Version: 0.10.0-snapshot

### Example flow

_No response_

### PR DESCRIPTION
**Pull request by Skraye**
close #1474

### CODE REVIEW

File: core/src/main/java/io/kestra/core/validations/ValidationFactory.java [hunk 1]
```diff
@@ -193,6 +192,16 @@ ConstraintValidator<FlowValidation, Flow> flowValidation() {
                 violations.add("Duplicate task id with name [" + String.join(", ", taskDuplicates) + "]");
             }
 
+            value.allTasksWithChilds()
+                .stream()
+                .forEach(
+                    task -> {
+                        if (task instanceof io.kestra.core.tasks.flows.Flow && ((io.kestra.core.tasks.flows.Flow) task).getFlowId().equals(value.getId())) {
+                            violations.add("Recursive call to flow [" + value.getId() + "]");
+                        }
```
Comment (loicmathieu) [inline core/src/main/java/io/kestra/core/validations/ValidationFactory.java:201]: You also need to check the namespace.

File: core/src/main/java/io/kestra/core/tasks/flows/Flow.java [hunk 2]
```diff
@@ -111,6 +111,12 @@ public Flow.Output run(RunContext runContext) throws Exception {
     public Execution createExecution(RunContext runContext, FlowExecutorInterface flowExecutorInterface) throws Exception {
         RunnerUtils runnerUtils = runContext.getApplicationContext().getBean(RunnerUtils.class);
 
+        Map<String, String> flowVars = (Map<String, String>) runContext.getVariables().get("flow");
+
+        if (flowVars.get("id") == ((Map<?, ?>) runContext.getVariables().get("flow")).get("id") && flowVars.get("namespace") == ((Map<?, ?>) runContext.getVariables().get("flow")).get("namespace")) {
+            throw new IllegalStateException("Recursive flow call detected");
+        }
+
```
Comment (loicmathieu) [inline core/src/main/java/io/kestra/core/tasks/flows/Flow.java:119]: If we cannot create a revursive flow, no need to check this at runtime.
Moreover it seems that this is always true!

**Review by loicmathieu — CHANGES_REQUESTED** [review]
Missing a test, you should create an invalid flow in the `flows/invalid` directory and test that executing this flow leads to an exception. Those test are located inside the FlowTest I think.

File: core/src/main/java/io/kestra/core/tasks/flows/Flow.java (hunk 2 above)
Comment (Skraye) [reply core/src/main/java/io/kestra/core/tasks/flows/Flow.java:119]: Dont we care about possibly existing recursive flow ?

Comment (loicmathieu) [reply core/src/main/java/io/kestra/core/tasks/flows/Flow.java:119]: I would say no, let's keep it simple for now

**Review by loicmathieu — APPROVED** [review]
LGTM

**Pull request merged**